### PR TITLE
Remove JUL logging support

### PR DESCRIPTION
**Pull request by jyemin**
The Java driver has supported logging with JUL (java.util.logging) since
its inception.  Since then, SLF4J has become the widely accepted logging
standard for Java and JUL has fallen out of favor. But for backwards
compatibility the driver has continued to fall back to JUL when SLF4J
classes are not found. This commit removes that fallback, making SLF4J
the only supported logging framework for the driver. If the
"org.sl4j.Logger" class is not found, the driver will print a single
warning to JUL and not log anything else for the duration of the
process.

https://jira.mongodb.org/browse/JAVA-2953

### CODE REVIEW

File: bson/src/main/org/bson/diagnostics/Logger.java [hunk 1]
```diff
@@ -34,15 +34,18 @@
      * @return True if this Logger is enabled for the TRACE level, false otherwise.
      * @since 1.4
      */
-    boolean isTraceEnabled();
+    default boolean isTraceEnabled() {
```
Comment (jyemin) [inline bson/src/main/org/bson/diagnostics/Logger.java]: Default method implementation make the NoOpLogger simpler

**Comment by rozza** [comment]
Do you think having a LoggerFactory setting would allow for better flexibility?  SLF4J is popular but so is Log4J and Apache Logging Commons?  I'm not sure how many people actually care about JUL but removing it may cause some issues.

**Comment by jyemin** [comment]
SLF4J provides bridges to all the popular logging frameworks so I think we're covered there.  And since there is a bridge to JUL itself (the inaptly named slf4j-jdk14 artifact), we're on pretty firm ground requiring SLF4J if you want logging.

We can always add a setting if there is demand for it.

**Comment by rozza** [comment]
Thanks for clarifying the SLF4J position, it makes sense to me rely on its bridge code to other loggers.

**Pull request closed without merging**